src/components/link.tsx

```diff
@@ -43,7 +44,8 @@ export class Link extends React.Component<LinkProps, {}> {
             toState,
             className,
             activeClassName,
-            children
+            children,
+            ...props
```

Comment:
Because `onClick` is not in this list, it will be passed to the `<a>` tag. This means that the local handler will not be called. Is this your intent?

Reply:
@nareshbhatia 
Yes. `onClick` will be in `...props` but it will be override in
```jsx
<a {...props} onClick={this.handleClick}/>
```

Reply:
Hi @rexpan, I am not in favor of this style of coding because it is very tricky to understand. If someone accidentally changes the order to the following, then `onClick` will no longer be overridden. Try it.

    <a onClick={this.handleClick} {...props}/>

I have added a `link.test.ts` file to master branch to help you experiment with this. Just uncomment the last test which is related to this PR. You will find that after reversing the order as above, `handleClick` will not be called at all!

In fact, it is not a good idea to duplicate properties in both places. The spread operator is meant to consume some properties and send others to the child component. See the discussion on [Spread Attributes](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes) in React docs. If we follow those recommendations, we should rename `...props` as `...other` and our code should look like this:

```
const {
    routerStore,
    toState,
    className,
    activeClassName,
    children,
    ...other
} = this.props;

...

return (
    <a
        className={joinedClassName}
        href={routerStateToUrl(routerStore, toState)}
        onClick={this.handleClick}
        {...other}
    >
        {children}
    </a>
);
```

In summary, I don't think supplying `onClick` as a prop to `Link` is a good idea. The very purpose of the `Link` component is to handle `onClick` in a certain way. What is the use case for handling it outside?

Reply:
@nareshbhatia 
I have update the PR, please check.
> What is the use case for handling it outside?

In my case, I need to log the event, do toggle (not active state) and handle other event like `focus`, `blur`.
Currently I work around using a `<div onClick={hClick} onFocus={hFocus}/>`, but it's kinda ugly and I also need to update the CSS.

Reply:
Looks good now! Thanks.